janus-idp/operator
# Optimize pod-refresh-on-external-config preprocessing and tests

## Description
This pull request:
- removes check for labels/annotations on hash calculation procedure avoiding extra creation of Deployment with checksum for "empty" config and -> unexpected additional Deployment update which should fix https://issues.redhat.com/browse/RHIDP-3217 (Backstage seems have the "concurrency" problem, starting up more than one instances)
- fixes config-refresh tests. Looks like newer Backstage version become more strict to the app-config content (it can not be "any" anymore) 
- adds retry.RetryOnConflict for better experience (retrying with no error messages in log) when add extra labels/annotations on CM/Secret


## Which issue(s) does this PR fix or relate to

https://issues.redhat.com/browse/RHIDP-3217

## PR acceptance criteria


## How to test changes / Special notes to the reviewer

1. Run CR with specified CM/Secrets, no extra Pod should appear on start time, no extra "CM/Secret version conflict" Log records show up  
2. integration_tests/config-refresh_test should pass.
